### ISSUE TITLE
separate "my-config" module

### ISSUE THREAD

**Issue by ray-holmium** [ne9z/dotfiles-flake#9]
While following the guide I was, for a time, slightly confused by the my-config module. It may be clearer/easier to follow for newcomers if there are separate modules for "user", "desktop" etc. This also provides the nice bonus of being able to toggle users and desktop independently per-host.

**Comment by ne9z** [comment]
This is the first feedback I received regarding "my-config".  I think it
would be fine to remove it completely, if it causes any confusion.

**Comment by ray-holmium** [comment]
In my case the confusion arose when I wanted to have a user with home-manager but I did not want gnome-desktop. I wasn't sure if I could remove my-config and start fresh on those modules without breaking anything else in the setup. So, your proposed solution would certainly help in that regard.

However, it is very convenient to have home-manager and gnome available on a new installation. In particular gnome is a crucial default for many users.

Perhaps a simple solution would be to further clarify the in-code comments re: my-config, to specify that the user can remove this module without breaking anything in the zfs-boot configuration. Please let me know if you would like to follow this course and I can examine the language to propose any changes that may have been helpful to me when I was first trying to figure this out.

This is a tricky issue because we are approaching the point where the line blurs between what is necessary for a clean install and what is just a nice-to-have on top.

In my own setup, I elected to duplicate "my-config" with one version acting as the module for user/home-manager and the other version as the module for gnome desktop.  If you were to adopt this, it would be a little more complicated than removal or further documenting, but I think this modular approach fits nicely with the guide's goal/ability to define multiple hosts. For example, in my setup, exampleHost/default.nix has the following nice clean toggles:

```
 users = {
    exampleUser.enable = true;
  };
  desktop = {
    gnome.enable = false;
  };
}
```
This allows me to define attributes globally for "exampleUser"  and then enable/disable him for any of the hosts in my setup. This foundation has been very helpful as I have expanded my setup to include more hosts, users, and desktop environments.

**Comment by ne9z** [comment]
I now gravitate to removing "my-config" altogether, but keep
"home-manager" in flake.nix for easy access.

Experienced users who know their way around Nix will know what to do.

Inexperienced new users are better off putting configuration in
"configuration.nix" directly rather than fiddling with custom options
and modules.  One can become reasonably proficient with Nix by reading
the Nix Pills series.

I will move GNOME to a comment in configuration.nix instead.  What do
you think?

**Comment by ray-holmium** [comment]
Agreed, this is an excellent solution. If an inexperienced user is reading official NixOS documentation, they will likely have an easier time interfacing with configuration.nix than with custom modules.

**Comment by ray-holmium** [comment]
Brilliant, thank you!
